met-web/src/components/survey/edit/EditForm.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useContext } from 'react';
+import { Grid, Stack, TextField } from '@mui/material';
+import { ActionContext } from './ActionContext';
+import { MetLabel, PrimaryButton, SecondaryButton } from 'components/common';
+import { SurveyFormProps } from '../types';
+
+export const EditForm = ({ handleClose }: SurveyFormProps) => {
+    const { handleSubmit, isSubmitting, submission, setSubmission } = useContext(ActionContext);
+
+    const handleChange = (value: string, commentIndex: number) => {
+        if (!submission) {
+            return;
+        }
+
+        submission.comments[commentIndex].text = value;
```

Comment:
React's convention is that we should always get create a copy of the state if we wanted to mutate it. Check this out https://stackoverflow.com/a/40309023